### DISCUSSION TITLE
What's the stack you're using ariadne in?

### DISCUSSION THREAD

**Discussion by pylipp** [mirumee/ariadne#1149] [Q&A]
Hej community, we from the [boxtribute project](https://github.com/boxwise/boxtribute) were wondering about the stack that people use ariadne in.
1. **Web framework**: do you use an async (FastAPI, Django, Starlette, Quart, sanic, ...) or a sync framework (Flask, Falcon, ...) and why?
2. **ORM**: do you use an ORM (Django, SQLAlchemy, Peewee, PonyORM, ...) and why did you make that choice?

To the mirumee folks @rafalp @patrys especially, can you share what stack works for you?

### Background
For the BE of our web application, we started with Flask-ariadne-Peewee. We had to integrate [batching via dataloaders](https://github.com/mirumee/ariadne/discussions/915#discussioncomment-3438068) at some point. We hence ended up with Flask running ariadne async including async dataloaders running sync Peewee ORM calls...
We'd like to be consistent :sweat_smile: 
- going pure sync would mean to use [this not very popular dataloader package](https://github.com/jkimbo/graphql-sync-dataloaders)
- going pure async would mean we can use subscriptions in the future; but we'd have to switch to frameworks like Quart (instead of Flask) and peewee-async.

**Comment by dacevedo12** [comment]
We're running full async in production

1. [starlette](https://github.com/encode/starlette) + [hypercorn](https://github.com/pgjones/hypercorn) (ASGI)
2. [aiodataloader](https://github.com/syrusakbary/aiodataloader) without ORM, [aioboto3](https://github.com/terrycain/aioboto3) + [aiobotocore](https://github.com/aio-libs/aiobotocore) ([aiohttp](https://github.com/aio-libs/aiohttp) under the hood)

Feel free to explore the setup at https://gitlab.com/fluidattacks/universe/-/tree/trunk/integrates/back/src/api

**Comment by rafalp** [comment]
Depending on a project we are using either Starlette, FastAPI or Mangum. For dataloader we are using aiodataloader.

Our Ariadne API's are proxies for microservices, so we are not using databases much in them. But when we do, its either encode/databases + Alembic, or SQLAlchemy 2 DBAL and Alembic.
